lib/topological_inventory/sync/inventory_upload/parser.rb

```diff
@@ -0,0 +1,46 @@
+module TopologicalInventory
+  class Sync
+    module InventoryUpload
+      class Parser
+        class << self
+          def parse_inventory_payload(url)
+            inventory = nil
+
+            open_url(url) do |io|
+              untargz(io) do |file|
+                require "json/stream"
+                inventory = JSON::Stream::Parser.parse(file)
+              end
+            end
+
+            inventory
+          end
+
+          private
+
+          def untargz(io)
+            require "rubygems/package"
+            Zlib::GzipReader.wrap(io) do |gz|
+              Gem::Package::TarReader.new(gz) do |tar|
```

Comment:
I'm thinking if we need to use Tar. Tar will join several files, but we wanted to send one compressed json schema, probably there is no situation we'll need archive with more files. We'll need most likely handle situation where we need to split data into more archives andsend them separatedly (insights-upload size limit)

Reply:
Tar gz is the standard format for the upload service